EngineLayer/CrosslinkSearch/CrosslinkSpectralMatch.cs

```diff
@@ -82,12 +82,27 @@ public void ResolveProteinPosAmbiguitiesForXl()
             }
         }
 
-        public static List<int> GetPossibleCrosslinkerModSites(char[] crosslinkerModSites, PeptideWithSetModifications peptide)
+        public static List<int> GetPossibleCrosslinkerModSites(char[] crosslinkerModSites, PeptideWithSetModifications peptide, bool CrosslinkAtCleavageSite)
         {
             List<int> possibleXlPositions = new List<int>();
             bool wildcard = crosslinkerModSites.Any(p => p == 'X');
 
-            for (int r = 0; r < peptide.BaseSequence.Length; r++)
+            //Consider the possibility that the site is at Protein N terminal.  
+            if (peptide.OneBasedStartResidueInProtein == 1)
+            {
+                if (!peptide.AllModsOneIsNterminus.Keys.Contains(1) && crosslinkerModSites.Contains(peptide.BaseSequence[0]) && CrosslinkAtCleavageSite)
+                {
+                    possibleXlPositions.Add(1);
+                }
+            }
```

Comment:
This is only checking the protein N-terminus, but not the C-terminus. Additionally, we're missing cases for variable methionine and endogenous cleavage products.